src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -72,6 +78,12 @@
      */
     private boolean ignoreOverridden;
 
+    /** The visibility scope where methods are checked. */
+    private Scope scope = Scope.ANONINNER;
```

Comment:
I also like the idea of `null` being the default and it means all scopes. Until I looked at the `Scope` class, I didn't understand how `ANONINNER` played in to the check. We should add to description and example that `null` for scope means all.